File: frontend/app/models/provider.ts
import { Model } from 'app/models/model';
import { Organisation } from 'app/models/organisation';
import { User } from 'app/models/user';

export class Provider extends Model {

	public admin: boolean;
	public approved: boolean;
	public organisation: Organisation;
	public user: User;
	public organisation_id: string;
	public user_id: string;

	constructor(json: any = {} as Provider) {
		super(json.id);
		this.admin = json.admin && json.admin || false;
		this.approved = json.approved && json.approved || false;
		this.organisation_id = json.organisation_id && json.organisation_id || '';
		this.user_id = json.user_id && json.user_id || '';
		this.organisation = json.organisation && new Organisation(json.organisation) || null;
		this.user = json.user && new User(json.user) || null;
	}
}

File: server/wupportal/src/main/java/de/codeschluss/wupportal/model/Provider.java
package de.codeschluss.wupportal.model;

import java.io.Serializable;
import java.util.List;

import javax.persistence.*;

import de.codeschluss.wupportal.model.Activity;
import de.codeschluss.wupportal.model.BaseEntity;
import de.codeschluss.wupportal.model.Organisation;

/**
 * The persistent class for the providers database table.
 * 
 */
@Entity
@Table(name = "providers")
public class Provider extends BaseEntity implements Serializable {
	private static final long serialVersionUID = 1L;

	private boolean admin;

	private boolean approved;

	@OneToMany(mappedBy = "provider")
	private List<Activity> activities;

	@ManyToOne
	private Organisation organisation;

	@ManyToOne
	private User user;

	public Provider() {
		super();
	}

	public boolean getAdmin() {
		return this.admin;
	}

	public void setAdmin(boolean admin) {
		this.admin = admin;
	}

	public boolean getApproved() {
		return this.approved;
	}

	public void setApproved(boolean approved) {
		this.approved = approved;
	}

	public List<Activity> getActivities() {
		return this.activities;
	}

	public void setActivities(List<Activity> activities) {
		this.activities = activities;
	}

	public Organisation getOrganisation() {
		return this.organisation;
	}

	public void setOrganisation(Organisation organisation) {
		this.organisation = organisation;
	}

	public User getUser() {
		return this.user;
	}

	public void setUser(User user) {
		this.user = user;
	}

}